### PR TITLE
Migrate DagFileProcessorManager._deactivate_stale_dags to Internal API

### LINKED ISSUE #28270: AIP-44 Migrate DagFileProcessorManager._deactivate_stale_dags to Internal API

### PR DESCRIPTION
**Pull request by vincbeck**
Migrate `DagFileProcessorManager._deactivate_stale_dags` to Internal API. Please while reviewing it to pay double attention to how I handle the logs. When discussing with @mhenc, we decided:
- If a function is NOT executed though internal API, the logger should stay as is
- If a function is executed through internal API, the logger should be separate

Very happy to hear your feedback and/or suggestions since, to be very honest, I am not a big fan of passing the logger as parameter

Closes #28270

### CODE REVIEW

**Comment by vincbeck** [comment]
cc @potiuk , @mhenc

File: airflow/api_internal/endpoints/rpc_api_endpoint.py [hunk 1]
```diff
@@ -68,7 +70,10 @@ def internal_airflow_api(
 
     log.debug("Calling method %.", {method_name})
     try:
-        output = handler(**params)
+        output = handler(
+            **params,
+            log=logging.getLogger(f"airflow.internal_api.{method_name}"),
```
Comment (mhenc) [inline airflow/api_internal/endpoints/rpc_api_endpoint.py]: Does it work if the method doesn't have "log" parameter?

File: airflow/dag_processing/manager.py [hunk 2]
```diff
@@ -485,51 +486,69 @@ def start(self):
         return self._run_parsing_loop()
 
     @provide_session
-    def _deactivate_stale_dags(self, session=None):
-        """
-        Detects DAGs which are no longer present in files.
-
-        Deactivate them and remove them in the serialized_dag table
-        """
+    def _scan_stale_dags(self, session=None):
```
Comment (mhenc) [inline airflow/dag_processing/manager.py]: I think we can remove the session param (and @provide_session) from this method.

**Review by mhenc — COMMENTED** [review]
Just two small nits. Thanks!

File: airflow/dag_processing/manager.py (hunk 2 above)
Comment (vincbeck) [reply airflow/dag_processing/manager.py]: Good point!

File: airflow/api_internal/endpoints/rpc_api_endpoint.py (hunk 1 above)
Comment (vincbeck) [reply airflow/api_internal/endpoints/rpc_api_endpoint.py]: Oh! That's a good point! I guess not. I dont think we should handle logs this way then. The simplest way would be to create a logger `log = logging.getLogger(__name__)` in `airflow/dag_processing/manager.py` and use this logger in static methods. The only downside is the logger would be the same whether or not internal API in turn on

Comment (mhenc) [reply airflow/api_internal/endpoints/rpc_api_endpoint.py]: I hoped that we can detect that the method has "log" argument and add it then.

If it's not possible to detect then  we can send addition parameter from client(like "inject_log=true") and when seen on server-side then add "log" to params

Comment (vincbeck) [reply airflow/api_internal/endpoints/rpc_api_endpoint.py]: I implemented a solution which detects if the parameter "log" exists. If it does, then provide one specific to internal API

Comment (mhenc) [reply airflow/api_internal/endpoints/rpc_api_endpoint.py]: This looks great! Thanks!

**Review by mhenc — APPROVED** [review]

**Comment by potiuk** [comment]
I guess that should be updated after discussion in https://github.com/apache/airflow/pull/28502 with `@classmethod` ?

**Comment by vincbeck** [comment]
> I guess that one should be updated after discussion in #28502 with `@classmethod` ?

Correct! I rather wait for #28502 to ber merged and then I'll update this PR

**Comment by potiuk** [comment]
> > I guess that one should be updated after discussion in #28502 with `@classmethod` ?
> 
> Correct! I rather wait for #28502 to be merged and then I'll update this PR

+1

**Comment by vincbeck** [comment]
PR updated :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**